Week-5/5.1/generatorFunctions.js

```diff
@@ -0,0 +1,39 @@
+async function doTask1() {
+    return new Promise(resolve => {
+		setTimeout(() => {
+			resolve(`Slept for: 2s`);
+            console.log(`Slept for: 2s`)
+		}, 2000)
+	})
+}
+async function doTask2() {
+    return new Promise(resolve => {
+		setTimeout(() => {
+			resolve(`Slept for: 3s`)
+            console.log(`Slept for: 3s`)
+
+		},3000)
+	})
+}
+async function doTask3() {
+    return new Promise(resolve => {
+		setTimeout(() => {
+			resolve(`Slept for: 4s`)
+            console.log(`Slept for: 4s`)
+
+		}, 4000)
+	})
+}
+function* mainGenerator(){
+    doTask1()
+    yield
```

Comment:
with yield add something, yield true or some value,
Also yield doTask1() is also valid